.travis.yml

```diff
@@ -22,27 +23,15 @@ addons:
             - pkg-config                   #needed for thrift cpp compilation
 
 before_install:
-    - pushd thirdparty
-    # thrift cpp
-    - >
-      if [ $TRAVIS_OS_NAME == linux ]; then
-        wget http://www.us.apache.org/dist/thrift/0.9.1/thrift-0.9.1.tar.gz &&
-        tar xfz thrift-0.9.1.tar.gz &&
-        pushd thrift-0.9.1 &&
-        ./configure --without-qt4 --without-c_glib --without-csharp --without-java --without-erlang --without-nodejs --without-lua --without-python --without-perl --without-php --without-php_extension --without-ruby --without-haskell --without-go --without-d --with-cpp --prefix=$HOME/local &&
-        make clean &&
-        make install &&
-        popd;
-      fi
-    - if [ $TRAVIS_OS_NAME == osx ]; then brew install thrift; fi
-    # snappy and lz4
-    - ./download_thirdparty.sh
-    - ./build_thirdparty.sh
-    - popd
+    - mkdir $HOME/build_dir
+    - cd $HOME/build_dir
+    - cp -r $TRAVIS_BUILD_DIR/thirdparty .
+    # thrift, snappy, and lz4
+    - ./thirdparty/download_thirdparty.sh
+    - ./thirdparty/build_thirdparty.sh
 
 before_script:
-    - mkdir build
-    - cd build
-    - THRIFT_HOME=$HOME/local cmake ..
+    - cd $HOME/build_dir
+    - THRIFT_HOME=$HOME/build_dir/thirdparty/installed/ SNAPPY_HOME=$HOME/build_dir/thirdparty/installed/ LZ4_HOME=$HOME/build_dir/thirdparty/installed/ cmake $TRAVIS_BUILD_DIR
```

Comment:
I think this is fine for now; at some point we can roll up this into a

./set-sandbox.env.sh

So if you want to use some other toolchain, you can do

./setup-my-toolchain.sh

which would be responsible for setting the right `FOO_HOME` variables (per the README instructions you added)